check if libyaml is already present before building it (take 2) (#45)
ros2/libyaml_vendor: README.md

# libyaml_vendor
CMake wrapper downloading and building libyaml.

This package CMake script check if a `libyaml` that has exactly the version required by ROS2 is available in the system. 

If such `libyaml` is available in the system, it only installs a [`Findyaml.cmake`](cmake/Modules/Findyaml.cmake) CMake find module, 
to ensure that the library can be always found via:
~~~cmake
find_package(yaml REQUIRED)
~~~
and linked via:
~~~cmake
target_link_libraries(<target> PRIVATE yaml)
~~~
even if no CMake config file for `yaml` is installed in the system.

If a `libyaml` that has exactly the version required by ROS2 is not available in the system, then it downloads and installs the 
`libyaml` as part of this package thanks to [CMake's ExternalProject module](https://cmake.org/cmake/help/latest/module/ExternalProject.html).

> The `FORCE_BUILD_VENDOR_PKG` CMake option (that by default is `OFF`) is provided to permit to force the building of `libyaml`, regardless

Reviewer: @traversaro nit: 
```suggestion
The `FORCE_BUILD_VENDOR_PKG` CMake option (that by default is `OFF`) is provided to allow forcing a `libyaml` build, regardless
```

Reviewer: @traversaro DCO isn't happy after applying the suggestion.

Quality declaration for this package: [libyaml_vendor QD](QUALITY_DECLARATION.md).

Quality declaration of external dependency [libyaml](./libyaml_q_declaration.md).

> 

Author: I was minimizing the changes w.r.t. to https://github.com/ros2/libyaml_vendor/pull/41 . Anyhow, I got rid of the change as it is seems that we are going forward with this PR.
